agent/getIn.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/**
+ * Retrieves the value from the path of nested objects
+ * @param  {Object} base Base or root object for path
+ * @param  {Array<String>} path nested path
+ * @return {any}      Value at end of path or `mull`
+ */
+function getIn(base, path) {
+  return path.reduce((obj, attr) => {
+    if (obj) {
+      if (obj.hasOwnProperty(attr)) {
+        return obj[attr];
+      }
+      if (typeof obj[Symbol.iterator] === 'function') {
+        // Convert iterable to array and return array[index]
+        return [...obj][attr];
```

Comment:
1. Won't there be n^2 complexity? getIn seems to be called in a loop, and we traversing the container every time it's called.
2. Will line 25 work with key-value maps, such as decorated records creating properties accessors for its members (so check in line #20 will fail)?